BoardConfiguration.cpp

```diff
@@ -47,125 +54,138 @@ void BoardConfiguration::generateRandomPositions() {
 }
 
 
-std::vector<std::string> BoardConfiguration::loadPositionsFromFile(std::string inputFile) {
-
-	vector<string> errorMessages;      // errorMessages.size() == 0
-
+void BoardConfiguration::loadPositionsFromFile(std::string inputFile) {
 	ifstream file(inputFile);    //open file for reading
 	if (!file.is_open()) {
-		errorMessages.push_back("Error opening file: " + inputFile);
-		return errorMessages;
+		_errors.push_back("Error opening file: " + inputFile);
+		return;
 	}
 
 	bool ifFinished = true;
-	map<char, int> wrongCharsMap;   //such as:  4,5,C,J,^,&, etc.
-	map<char, int> validationToolsMap;  //settings validation for players A, B
-	initializeValidationToolsMap(validationToolsMap);
-
-	string line = readLineFromFile(file);   //read a whole line until \n or \r\n  or EOF
-	while (!file.eof() && ifFinished) {
-		if (updatePositions(line, wrongCharsMap, validationToolsMap) <= ROWS) {      //read 13 lines
-			string line = readLineFromFile(file);
-		}
-		else {
-			ifFinished = false;
+	initializeValidationMaps();
+
+	int rowCount = 0;
+	while (rowCount < ROWS) {
```

Comment:
אם יש לך גם משתנה מאותחל, גם אתה יודע עד מתי הוא אמור להסתיים, וגם אתה מקדם אותו אותו דבר כל הזמן,
לא עדיף לעשות for ??